prover/prover_cli/src/commands/restart/batch.rs

```diff
@@ -0,0 +1,221 @@
+use anyhow::{bail, Context};
+use clap::Args as ClapArgs;
+use prover_dal::{
+    fri_witness_generator_dal::FriWitnessJobStatus, Connection, ConnectionPool, Prover, ProverDal,
+};
+use zksync_config::PostgresConfig;
+use zksync_env_config::FromEnv;
+use zksync_types::{basic_fri_types::AggregationRound, L1BatchNumber};
+
+#[derive(ClapArgs)]
+pub(crate) struct Args {
+    /// Batch number to restart.
+    #[clap(short = 'n')]
+    batch: L1BatchNumber,
+    /// Restart all prover jobs of the batch.
+    #[clap(short, long, requires = "batch", conflicts_with_all = ["basic_witness_generator", "leaf_witness_generator", "node_witness_generator", "recursion_tip", "scheduler", "compressor"])]
+    prover_jobs: Option<bool>,
+    /// Aggregation round from which you want to reset the prover jobs.
+    #[clap(short, long, requires = "prover_jobs", conflicts_with_all = ["basic_witness_generator", "leaf_witness_generator", "node_witness_generator", "recursion_tip", "scheduler", "compressor"])]
+    aggregation_round: Option<AggregationRound>,
+    /// Restart the basic witness generator job of the batch.
+    #[clap(long, alias = "bwg", requires = "batch", conflicts_with_all = ["prover", "leaf_witness_generator", "node_witness_generator", "recursion_tip", "scheduler", "compressor"])]
+    basic_witness_generator: Option<bool>,
+    /// Restart the leaf witness generator jobs of the batch.
+    #[clap(long, alias = "lwg", requires = "batch", conflicts_with_all = ["prover", "basic_witness_generator", "node_witness_generator", "recursion_tip", "scheduler", "compressor"])]
+    leaf_witness_generator: Option<bool>,
+    /// Restart the node witness generator jobs of the batch.
+    #[clap(long, alias = "nwg", requires = "batch", conflicts_with_all = ["prover", "basic_witness_generator", "leaf_witness_generator", "recursion_tip", "scheduler", "compressor"])]
+    node_witness_generator: Option<bool>,
+    /// Restart the recursion tip job of the batch.
+    #[clap(long, alias = "rt", requires = "batch", conflicts_with_all = ["prover", "basic_witness_generator", "leaf_witness_generator", "node_witness_generator", "scheduler", "compressor"])]
+    recursion_tip: Option<bool>,
+    /// Restart the scheduler job of the batch.
+    #[clap(short, long, requires = "batch", conflicts_with_all = ["prover", "basic_witness_generator", "leaf_witness_generator", "node_witness_generator", "recursion_tip", "compressor"])]
+    scheduler: Option<bool>,
+    /// Restart the compressor job of the batch.
+    #[clap(short, long, requires = "batch", conflicts_with_all = ["basic_witness_generator", "leaf_witness_generator", "node_witness_generator", "recursion_tip", "scheduler"])]
+    compressor: Option<bool>,
+}
+
+pub(crate) async fn run(args: Args) -> anyhow::Result<()> {
+    let config = PostgresConfig::from_env()?;
+    let prover_connection_pool = ConnectionPool::<Prover>::singleton(config.prover_url()?)
```

Comment:
I think we can name it just connection_pool, I don't think we are trying to touch core DB here